testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/updaters/ServerResourceUpdater.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.testsuite.updaters;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+/**
+ *
+ * @author hmlnarik
+ */
+public abstract class ServerResourceUpdater<T extends ServerResourceUpdater, Res, Rep> implements Closeable {
+
+    protected final Res resource;
+    protected final Rep rep;
+    protected final Rep origRep;
+    protected Consumer<Rep> updater;
+    protected boolean updated = false;
+
+    public ServerResourceUpdater(Res resource, Supplier<Rep> representationGenerator, Consumer<Rep> updater) {
+        this.resource = resource;
+        this.updater = updater;
+
+        // origRep and rep need to be two different instances
+        this.origRep = representationGenerator.get();
+        this.rep = representationGenerator.get();
+    }
+
+    public Res getResource() {
+        return resource;
+    }
+
+    public T update() {
+        performUpdate(origRep, rep);
+        this.updated = true;
+        return (T) this;
+    }
+
+    protected void performUpdate(Rep from, Rep to) {
+        updater.accept(rep);
+    }
+
+    public T updateWith(Consumer<Rep> representationUpdater) {
+        representationUpdater.accept(this.rep);
+        return (T) this;
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (! this.updated) {
+            throw new IOException("Attempt to revert changes that were never applied - have you called " + this.getClass().getName() + ".update()?");
+        }
+        performUpdate(rep, origRep);
```

Comment:
Isn't this a typo? Is it necessary to call performUpdate at close?

Reply:
Ah, I see. Makes sense together with the other change in performUpdate :) Thanks!